### PR TITLE
UI: Show correct nav items when in chroot namespace

### PR DESCRIPTION
**Pull request by hashishaw**
This PR handles the necessary updates to enable a user logged in on a chrooted namespace listener to see their correct nav items. This depends on backend work in 1.16 ([backported to 1.15.5](https://github.com/hashicorp/vault/pull/24383)) which adds `chroot_namespace` key to the `resultant-acl` endpoint.

### CODE REVIEW

File: ui/tests/unit/services/permissions-test.js [hunk 1]
```diff
@@ -53,66 +53,11 @@ module('Unit | Service | permissions', function (hooks) {
     assert.deepEqual(this.service.get('globPaths'), PERMISSIONS_RESPONSE.data.glob_paths);
   });
 
-  test('returns true if a policy includes access to an exact path', function (assert) {
```
Comment (hashishaw) [inline ui/tests/unit/services/permissions-test.js:56]: Reorganized these tests, so these are just moved to a module now

File: ui/tests/unit/services/permissions-test.js [hunk 2]
```diff
@@ -148,77 +93,161 @@ module('Unit | Service | permissions', function (hooks) {
...
+  module('pathWithNamespace', function () {
+    test('appends the namespace to the path if there is one', function (assert) {
+      const namespaceService = Service.extend({
+        path: 'marketing',
+      });
+      this.owner.register('service:namespace', namespaceService);
+      assert.strictEqual(this.service.pathNameWithNamespace('sys/auth'), 'marketing/sys/auth');
+    });
+
+    test('appends the chroot and namespace when both present', function (assert) {
```
Comment (hashishaw) [inline ui/tests/unit/services/permissions-test.js:228]: This test and the following ones are new

File: ui/app/components/sidebar/nav/cluster.js [hunk 3]
```diff
@@ -15,4 +15,9 @@ export default class SidebarNavClusterComponent extends Component {
   get cluster() {
     return this.currentCluster.cluster;
   }
+
+  get isRootNamespace() {
+    // should only return true if we're in the true root namespace
```
Comment (kiannaquach) [inline ui/app/components/sidebar/nav/cluster.js:20]: ty for the comment!

Comment (hellobontempo) [reply ui/app/components/sidebar/nav/cluster.js:20]: I'm still getting up to speed with the `chroot` situation - does true "root" mean `root` root? 

Does `this.currentCluster.hasChrootNamespace` mean the user has a configured root namespace and so they will not have `root` as the base/parent namespace?

File: ui/tests/unit/services/permissions-test.js [hunk 4]
```diff
@@ -148,77 +93,161 @@ module('Unit | Service | permissions', function (hooks) {
...
+    test('handles superfluous slashes', function (assert) {
+      const namespaceService = Service.extend({
+        path: '/marketing',
+      });
+      this.owner.register('service:namespace', namespaceService);
+      this.service.set('chrootNamespace', '/admin/');
+      assert.strictEqual(this.service.pathNameWithNamespace('/sys/auth'), 'admin/marketing/sys/auth');
+      assert.strictEqual(
+        this.service.pathNameWithNamespace('/sys/policies/'),
+        'admin/marketing/sys/policies/'
```
Comment (hellobontempo) [inline ui/tests/unit/services/permissions-test.js:249]: is this expected to have the path end in a `/`?

**Review by hellobontempo — APPROVED** [review]
Mostly clarifying questions! 🚢

File: ui/app/components/sidebar/nav/cluster.js (hunk 3 above)
Comment (hashishaw) [reply ui/app/components/sidebar/nav/cluster.js:20]: `hasChrootNamespace` means the Vault operator has set `chroot_namespace` to some value on the config of the given listener, and so the user does not have access to "true root" (which yes, is `root` namespace). So, even if the UI is in its top-most namespace, if `chroot_namespace` is configured it is not the true root. 

Does that answer your question?

File: ui/tests/unit/services/permissions-test.js (hunk 4 above)
Comment (hashishaw) [reply ui/tests/unit/services/permissions-test.js:249]: Yes, for glob paths they end in `/`

File: ui/app/components/sidebar/nav/cluster.js (hunk 3 above)
Comment (hellobontempo) [reply ui/app/components/sidebar/nav/cluster.js:20]: Yes! I think so - I wanted to clarify that setting the `chroot_namespace` means that you are configuring the top namespace and so therefore `root` will NOT be a possible/accessible. 

Thank you!

**Pull request merged**